File: Models/EventModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace t5_back.Models;
public class Event
{
    [Key]
    public Guid UUID { get; set; }
    
    [Required]
    [MaxLength(255)]
    public string Title { get; set; } = string.Empty;
    
    [Required]
    public DateTime StartDate { get; set; }
    
    [Required]
    public DateTime EndDate { get; set; }
    
    [Required]
    public EventStatus Status { get; set; }

    [Required]
    public int MinDurationMinutes { get; set; }

    [Required]
    public int MaxDurationMinutes { get; set; }
    
    [JsonIgnore]
    public ICollection<Team>? Teams { get; set; }
    
    [JsonIgnore]
    public ICollection<Point>? Points { get; set; }
    
    [JsonIgnore]
    public ICollection<Area>? Areas { get; set; }
    
    [JsonIgnore]
    public ICollection<RoutePath>? Paths { get; set; }
    
    [JsonIgnore]
    public ICollection<SecurityZone>? SecurityZones { get; set; }
}

File: frontend/Nidhoggr_front/src/app/models/eventModel.ts
export enum EventStatus {
  ToOrganize = 0,
  InProgress = 1,
  Installation = 2,
  Uninstallation = 3,
  Completed = 4
}

export interface Event {
  uuid: string;
  title: string;
  startDate: Date;
  endDate: Date;
  status: EventStatus;
  minDurationMinutes?: number;
  maxDurationMinutes?: number;
  isArchived?: boolean;
  isFavorite?: boolean;
}
